File: src/test/java/com/circleescape/server/model/GreedyGuardTest.java
package com.circleescape.server.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class GreedyGuardTest {

	@Test
	void test() {
		GuardStrategy gs = new GreedyGuard();
		GameParameters parameters = new GameParameters(100, 4);
		double nextDuck = gs.updateCat(parameters, MathUtils.TAU *12/14, duck100(MathUtils.TAU * 13/14), duck100(MathUtils.TAU * 1/14));

		System.out.println(MathUtils.TAU * 1/14);
		System.out.println(nextDuck);
		assertTrue(Math.abs(nextDuck - MathUtils.TAU * 1/14) < MathUtils.ϵ);
	}
	
	private PolarCoordinates duck100(double rad) {
		return new PolarCoordinates(100, rad);
	}

}

File: src/test/kotlin/com/circleescape/server/model/GreedyGuardTest.kt
package com.circleescape.server.model

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class GreedyGuardTest {

    @Test
    fun test() {
        val gs: GuardStrategy = GreedyGuard()
        val parameters = GameParameters(100.0, 4.0)
        val nextDuck = gs.updateCat(
            parameters, MathUtils.TAU * 12 / 14, duck100(MathUtils.TAU * 13 / 14), duck100(
                MathUtils.TAU * 1 / 14
            )
        )
        println(MathUtils.TAU * 1 / 14)
        println(nextDuck)
        Assertions.assertTrue(Math.abs(nextDuck - MathUtils.TAU * 1 / 14) < MathUtils.ϵ)
    }

    private fun duck100(rad: Double): PolarCoordinates {
        return PolarCoordinates(100.0, rad)
    }
}